app/views/devise/sessions/new.html.erb

```diff
@@ -1,19 +1,38 @@
-<h2 class="govuk-heading-l">Log in</h2>
+  <% if session[:challenge_name].present? %>
+    <h2 class="govuk-heading-l">Log in - One Time Password Request</h2>
 
-<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
-  <div class="govuk-form-group">
-    <%= f.label :email, class: "govuk-label" %>
-    <%= f.email_field :email, autofocus: true, autocomplete: "email", class: "govuk-input" %>
-  </div>
+    <p class="govuk-body-l">Please enter your one time code from
+    <%= session[:challenge_parameters]["FRIENDLY_DEVICE_NAME"] %>.</p>
 
-  <div class="govuk-form-group">
-    <%= f.label :password, class: "govuk-label" %>
-    <%= f.password_field :password, autocomplete: "current-password", class: "govuk-input" %>
-  </div>
+    <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
```

Comment:
too much duplication here... the `form_for` doesn't need to be duplicated anymore, same with the submit button etc...

Reply:
Changes made...